From 086798e9dd3c2cbec1f3795c21425edb0bb9b805 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 12 Dec 2021 00:54:41 +0000
Subject: [PATCH 3/6] gnu: Add go-textseg v13

* gnu/packages/golang.scm: (go-github-com-apparentlymart-go-textseg-v13,
go-github-com-apparentlymart-go-textseg-autoversion): New variables.
---
 gnu/packages/golang.scm | 39 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 616a86cb99..566be61eec 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -70,6 +70,7 @@ (define-module (gnu packages golang)
   #:use-module (gnu packages perl)
   #:use-module (gnu packages pkg-config)
   #:use-module (gnu packages pulseaudio)
+  #:use-module (gnu packages ruby)
   #:use-module (gnu packages syncthing)
   #:use-module (gnu packages terminals)
   #:use-module (gnu packages textutils)
@@ -8973,3 +8974,41 @@ (define-public go-github-com-vmihailenco-msgpack-v4
      "This package provides implementation of MessagePack encoding for Go
 programming language")
     (license license:bsd-2)))
+
+(define-public go-github-com-apparentlymart-go-textseg-v13
+  (package
+    (name "go-github-com-apparentlymart-go-textseg-v13")
+    (version "13.0.0")
+    (source
+      (origin
+        (method git-fetch)
+        (uri (git-reference
+               (url "https://github.com/apparentlymart/go-textseg")
+               (commit (string-append "v" version))))
+        (file-name (git-file-name name version))
+        (sha256
+          (base32 "0gdgi0d52rq1xsdn9icc8lghn0f2q927cifmrlfxflf7bf21vism"))))
+    (build-system go-build-system)
+    (arguments
+     '(#:unpack-path "github.com/apparentlymart/go-textseg/v13"
+       #:import-path "github.com/apparentlymart/go-textseg/v13/textseg"))
+    (native-inputs
+     `(("ruby" ,ruby)))
+    (home-page "https://github.com/apparentlymart/go-textseg")
+    (synopsis "Go implementation of Unicode Text Segmentation")
+    (description
+     "This package provides an implementation of the Unicode Text Segmentation
+specification for Go.  Specifically, it currently includes only the grapheme
+cluster segmentation algorithm.")
+    ;; TODO: (Sharlatan-20211211T234916+0000): some not clear for me mixture of
+    ;; licenses, need verification
+    ;; https://github.com/apparentlymart/go-textseg/blob/master/LICENSE
+    (license #f)))
+
+(define-public go-github-com-apparentlymart-go-textseg-autoversion
+  (package
+    (inherit go-github-com-apparentlymart-go-textseg-v13)
+    (name "go-github-com-apparentlymart-go-textseg-autoversion")
+    (arguments
+     '(#:unpack-path "github.com/apparentlymart/go-textseg/autoversion"
+       #:import-path "github.com/apparentlymart/go-textseg/autoversion/textseg"))))
-- 
2.34.0

